lib/iris/coords.py

```diff
@@ -1191,27 +1185,16 @@ def serialize(x):
                 warnings.warn(msg.format(self.name()))
 
             # Create bounds for the new collapsed coordinate.
-            item = self.core_bounds() if self.has_bounds() \
+            item = np.concatenate(self.core_bounds()) if self.has_bounds() \
                 else self.core_points()
-            lower, upper = item.min(), item.max()
-            bounds_dtype = item.dtype
-            # Ensure 2D shape of new bounds.
-            bounds = np.empty((1, 2), 'object')
-            bounds[0, 0] = lower
-            bounds[0, 1] = upper
-            # Create points for the new collapsed coordinate.
-            points_dtype = self.dtype
-            points = (float(lower) + float(upper)) * 0.5
+
+            # Calculate the bounds and points along the right dims
+            bounds = np.stack([item.min(axis=dims_to_collapse),
+                               item.max(axis=dims_to_collapse)]).T
+            points = item.mean(axis=dims_to_collapse, dtype=self.dtype)
 
             # Create the new collapsed coordinate.
-            if is_lazy_data(item):
```

Comment:
I'm not convinced that we aren't now losing some lazy behaviour here.
As this is using np.concatenate / np.stack in place of Dask ones.
Otherwise what was `multidim_lazy_stack` doing ?
I suspect we need another test, that doesn't currently exist, to ensure that collapse is 'lazy' in these cases.

Reply:
Yes - good point, thanks